src/snarky/parties-leaves.ts

```diff
@@ -12,8 +12,8 @@ export {
 
 export { toJson, toJsonLeafTypes, toFields, toFieldsLeafTypes };
 
-type UInt64 = { value: Field };
-type UInt32 = { value: Field };
+type UInt64 = { value: Field; type: 'UInt64' };
```

Comment:
Unfortunately doesn't seem to solve my use case.. I need `UInt64` to not extend `UInt32` and vice versa.

I tried the following snippet:
```ts
type FieldLike<N> = { value: Field };
type UInt64 = FieldLike<{ type: 'UInt64' }>;
type UInt32 = FieldLike<{ type: 'UInt32' }>;
type X = UInt32 extends UInt64 ? 'not working' : 'it works!';
```
For me, `X` is of type `"not working"`..

Reply:
I think my original solution can be made slightly better by making the type prop optional:

```ts
type UInt64 = { value: Field; type?: 'UInt64' };
type UInt32 = { value: Field; type?: 'UInt32' };
```
that way, they don't extend each other and we can just instantiate them without the `type` field at runtime.

it's still not perfect because the `type` field shows up in intellisense

Reply:
That's weird that they extend each other, shows that I do not understand typescript's type system enough.

I think this is good with them optional -- since it appears in intellisense can you use some kind of prefix so people know to avoid it: like `_type` or `$type` (I'm not sure if we have a standard for this yet)